plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlScanner.java

```diff
@@ -40,6 +60,65 @@ public void startElement(String uri, String localName, String qName, Attributes
         addToBlacklist(new JavaServletElement(Messages.getString("web.xml.version"), start, 0));
       }
     }
+    if ("servlet-class".equalsIgnoreCase(localName)) {
+      Locator2 locator = getLocator();
+      insideServletClass = true;
+      servletClassContents = new StringBuilder();
+      servletClassLocation = new DocumentLocation(locator.getLineNumber(),
+          locator.getColumnNumber() - localName.length() - 2);
+    }
+  }
+  
+  @Override
+  public void characters(char ch[], int start, int length) throws SAXException {
+    if (insideServletClass) {
+      servletClassContents.append(ch, start, length);
+    }
+  }
+  
+  @Override
+  public void endElement(String uri, String localName, String qName) throws SAXException {
+    if ("servlet-class".equalsIgnoreCase(localName)) {
+      insideServletClass = false;
+      String servletClassName = servletClassContents.toString();
+      if (findClass(servletClassName) == null) {
+        addToBlacklist(new UndefinedServletElement(
+            servletClassName, servletClassLocation, localName.length() + 2));
+      }
+    }
+  }
+  
+  /**
+   * Checks for the given class name within the project.
+   */
+  static IType findClass(String typeName) {
+    if (typeName == null || "".equals(typeName)) {
```

Comment:
Could use Guava's `String.isNullOrEmpty()` if Guava is already a dependency of this bundle.